spyder/plugins/projects/widgets/main_widget.py

```diff
@@ -91,18 +724,148 @@ def set_project_dir(self, directory):
+
+        Returns
+        -------
+        spyder.plugins.projects.api.BaseProjectType
+            Loaded project type class.
+        """
+        fpath = osp.join(
+            path, get_project_config_folder(), 'config', WORKSPACE + ".ini")
+
+        project_type_id = EmptyProject.ID
+        if osp.isfile(fpath):
+            config = configparser.ConfigParser()
+
+            # Catch any possible error when reading the workspace config file.
+            # Fixes spyder-ide/spyder#17621
+            try:
+                config.read(fpath, encoding='utf-8')
+            except Exception:
+                pass
+
+            # This is necessary to catch an error for projects created in
+            # Spyder 4 or older versions.
+            # Fixes spyder-ide/spyder#17097
+            try:
+                project_type_id = config[WORKSPACE].get(
+                    "project_type", EmptyProject.ID)
+            except Exception:
+                pass
+
+        EmptyProject._PARENT_PLUGIN = self.get_plugin()
+        project_types = self.get_project_types()
+        project_type_class = project_types.get(project_type_id, EmptyProject)
+        return project_type_class
+
+    def _add_to_recent(self, project):
+        """
+        Add an entry to recent projetcs
+
+        We only maintain the list of the 10 most recent projects
+        """
+        if project not in self.recent_projects:
+            self.recent_projects.insert(0, project)
+
+        if len(self.recent_projects) > self.get_conf('max_recent_projects'):
+            self.recent_projects.pop(-1)
+
+    def _setup_menu_actions(self):
+        """Setup and update the menu actions."""
+        if self.recent_projects:
+            for project in self.recent_projects:
+                if self.is_valid_project(project):
+                    if os.name == 'nt':
+                        name = project
+                    else:
+                        name = project.replace(get_home_dir(), '~')
+
+                    try:
+                        action = self.get_action(name)
+                    except KeyError:
+                        action = self.create_action(
+                            name,
+                            text=name,
+                            icon=self.create_icon('project'),
+                            triggered=self._build_opener(project),
+                        )
+
+                    self.add_item_to_menu(
+                        action,
+                        menu=self.recent_project_menu,
+                        section=RecentProjectsMenuSections.Recent)
+
+        for item in [
+                self.clear_recent_projects_action,
+                self.max_recent_action]:
+            self.add_item_to_menu(
+                item,
+                menu=self.recent_project_menu,
+                section=RecentProjectsMenuSections.Extras)
+
+        self._update_project_actions()
```

Comment:
Not totally sure why but seems like something strange is happening when building the recent projects entry?:

Checked creating a project, closing it and reopening and I got this layout for the recent projects menu:

![recent_projects](https://user-images.githubusercontent.com/16781833/232915109-965508a6-1a83-4b02-8b8d-08b7cf708931.png)


Besides the duplicated entry, should not the actions for maximum number and to clear the list be below the actual recent projects entries?

Reply:
Closed and open again Spyder and after that the actions order was reverted:

 
![recent_projects_2](https://user-images.githubusercontent.com/16781833/232917541-654626c9-f144-4984-91f4-bfe40046a5c8.png)

Reply:
Good catches @dalthviz! I'm sure they also affect `5.x` (because I only moved code around) and I think I fixed them in my last commits, so please test again.